src/Sidebar/ui/SidebarItem.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { memo } from "react";
+import Badge from "@mui/material/Badge";
+import ListItem from "@mui/material/ListItem";
+import ListItemIcon from "@mui/material/ListItemIcon";
+import ListItemText from "@mui/material/ListItemText";
+import Tooltip from "@mui/material/Tooltip";
+import Typography from "@mui/material/Typography";
+
+import type { Page } from "../../ui/Router";
+
+export interface ICreateProps {
+  key_: Page;
+  icon: React.ReactElement;
+  count?: number;
+  active?: boolean;
+}
+
+export interface IProps extends ICreateProps {
+  clickFn: () => void;
+  flash: boolean;
+  classes: any;
+  sidebarOpen: boolean;
+}
+
+export const SidebarItem = memo(function (props: IProps): React.ReactElement {
+  // Obnoxious, because we can't modify props at all.
+  const icon = Object.assign({}, props.icon);
+  icon.props = Object.assign({ color: props.flash ? "error" : !props.active ? "secondary" : "primary" }, icon.props);
```

Comment:
```suggestion
  const icon = {
    type: props.icon.type,
    key: props.icon.key,
    props: { ...props.icon.props, color: props.flash ? "error" : props.active ? "primary" : "secondary" },
  };
```
Avoids `any` type from `Object.assign`, and should also be a more efficient way of copying the icon object with a modified color.

Reply:
Actually better solution is to not send the built jsx icon as a prop at all and just send the function for creating a new icon of the given type, I will send a different suggestion later this morning that avoids this whole thing where we break down a prop icon to rebuild a clone and display the clone.

Reply:
Done, and rebased/ready to go

Reply:
> Actually better solution is to not send the built jsx icon as a prop at all and just send the function for creating a new icon of the given type, I will send a different suggestion later this morning that avoids this whole thing where we break down a prop icon to rebuild a clone and display the clone.

This doesn't work, unless you know of a snazzy way to package other props with it. That was my initial attempt, but it foundered on the situation with `doubleArrowIcon`.
You could just send the props I suppose, but the syntax of specifying an icon as JSX is nicer.